### PR TITLE
[chore][exporter/awsemfexporter] Add metadata.yaml

### PR DESCRIPTION
**Pull request by bryan-aguilar**
**Description:** add metadata.yaml
**Link to tracking Issue:** #21213

### CODE REVIEW

File: exporter/awsemfexporter/metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+type: awsemf
+
+status:
+  class: exporter
+  stability:
+    beta: [metrics]
+  distributions: [contrib, aws]
```
Comment (bryan-aguilar) [inline exporter/awsemfexporter/metadata.yaml:7]:
```suggestion
  distributions: [contrib, aws]

```

**Comment by MovieStoreGuy** [comment]
Looks like a potential flake? Could I ask that you rebase against main.

File: exporter/awsemfexporter/README.md [hunk 2]
```diff
@@ -1,10 +1,15 @@
...
-| Stability                | [beta]    |
-| Supported pipeline types |  metrics  |
-| Distributions            | [contrib] |
+<!-- status autogenerated section -->
+| Status        |           |
+| ------------- |-----------|
+| Stability     | [beta]: metrics   |
+| Distributions | [contrib], [aws] |
+
+[beta]: https://github.com/open-telemetry/opentelemetry-collector#beta
```
Comment (codeboten) [inline exporter/awsemfexporter/README.md:9]: The links at the bottom can be removed now

File: exporter/awsemfexporter/doc.go [hunk 3]
```diff
@@ -14,4 +14,6 @@
 
 // Package awsemfexporter implements an OpenTelemetry Collector exporter that sends EmbeddedMetricFormat to
 // AWS CloudWatch Logs in the region the collector is running in using the PutLogEvents API.
+//
+//go:generate mdatagen metadata.yaml
```
Comment (codeboten) [inline exporter/awsemfexporter/doc.go:18]: for consistency w/ the other packages, please leave the docstring above the package definition

File: exporter/awsemfexporter/factory.go [hunk 4]
```diff
@@ -22,23 +22,17 @@ import (
 	"go.opentelemetry.io/collector/exporter/exporterhelper"
 	"go.uber.org/zap"
 
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/awsemfexporter/internal/metadata"
```
[the remark this answers is not in the document]
Comment (bryan-aguilar) [reply exporter/awsemfexporter/factory.go:25]: Intellij IDE using the `goimports` auto formatting for imports forces this order. Is there something here that I can change to force this? Also, don't we have a `goimports` linter for this? Should that be failing if this is the case?

**Review by codeboten — APPROVED** [review]

**Pull request merged**